perfkitbenchmarker/benchmarks/oldisim_benchmark.py

```diff
@@ -0,0 +1,253 @@
+# Copyright 2015 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Runs oldisim.
+
+oldisim is a framework to support benchmarks that emulate Online Data-Intensive
+(OLDI) workloads, such as web search and social networking. oldisim includes
+sample workloads built on top of this framework.
+
+With its default config, oldisim models an example search topology. A user query
+is first processed by a front-end server, which then eventually fans out the
+query to a large number of leaf nodes. The latency is measured at the root of
+the tree, and often increases with the increase of fan-out. oldisim reports a
+scaling efficiency for a given topology. The scaling efficiency is defined
+as queries per second (QPS) at the current fan-out normalized to QPS at fan-out
+1 with ISO root latency.
+
+Sample command line:
+
+./pkb.py --benchmarks=oldisim --project='YOUR_PROJECT' --num_leaves=4
+--fanout=1,2,3,4 --latency_target=40 --latency_metric=avg
+
+The above command will build a tree with one root node and four leaf nodes. The
+average latency target is 40ms. The root node will vary the fanout from 1 to 4
+and measure the scaling efficiency.
+"""
+
+import logging
+import os
+import re
+import time
+
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import sample
+from perfkitbenchmarker import vm_util
+
+
+FLAGS = flags.FLAGS
+
+flags.DEFINE_integer('num_leaves', 4, 'number of leaf nodes',
```

Comment:
Would you mind prefixing these flags with the benchmark name (e.g. `oldisim_num_leaves`), so it's clear what benchmark they affect?